content/300-guides/100-performance-and-optimization/090-metrics.mdx

```diff
@@ -0,0 +1,493 @@
+
+# HELP pool_wait_duration_ms The wait time for a worker to get a connection.
+# TYPE pool_wait_duration_ms histogram
+pool_wait_duration_ms_bucket{le="0"} 0
+pool_wait_duration_ms_bucket{le="1"} 3
+pool_wait_duration_ms_bucket{le="5"} 3
+pool_wait_duration_ms_bucket{le="10"} 3
+pool_wait_duration_ms_bucket{le="50"} 3
+pool_wait_duration_ms_bucket{le="100"} 3
+pool_wait_duration_ms_bucket{le="500"} 3
+pool_wait_duration_ms_bucket{le="1000"} 3
+pool_wait_duration_ms_bucket{le="5000"} 3
+pool_wait_duration_ms_bucket{le="50000"} 3
+pool_wait_duration_ms_bucket{le="+Inf"} 3
+pool_wait_duration_ms_sum 0.023208
+pool_wait_duration_ms_count 3
+
+# HELP query_operation_total_elapsed_time_ms
+# TYPE query_operation_total_elapsed_time_ms histogram
+query_operation_total_elapsed_time_ms_bucket{le="0"} 0
+query_operation_total_elapsed_time_ms_bucket{le="1"} 1
+query_operation_total_elapsed_time_ms_bucket{le="5"} 2
+query_operation_total_elapsed_time_ms_bucket{le="10"} 2
+query_operation_total_elapsed_time_ms_bucket{le="50"} 2
+query_operation_total_elapsed_time_ms_bucket{le="100"} 2
+query_operation_total_elapsed_time_ms_bucket{le="500"} 2
+query_operation_total_elapsed_time_ms_bucket{le="1000"} 2
+query_operation_total_elapsed_time_ms_bucket{le="5000"} 2
+query_operation_total_elapsed_time_ms_bucket{le="50000"} 2
+query_operation_total_elapsed_time_ms_bucket{le="+Inf"} 2
+query_operation_total_elapsed_time_ms_sum 3.197624
+query_operation_total_elapsed_time_ms_count 2
+
+# HELP query_total_elapsed_time_ms
+# TYPE query_total_elapsed_time_ms histogram
+query_total_elapsed_time_ms_bucket{le="0"} 0
+query_total_elapsed_time_ms_bucket{le="1"} 5
+query_total_elapsed_time_ms_bucket{le="5"} 5
+query_total_elapsed_time_ms_bucket{le="10"} 5
+query_total_elapsed_time_ms_bucket{le="50"} 5
+query_total_elapsed_time_ms_bucket{le="100"} 5
+query_total_elapsed_time_ms_bucket{le="500"} 5
+query_total_elapsed_time_ms_bucket{le="1000"} 5
+query_total_elapsed_time_ms_bucket{le="5000"} 5
+query_total_elapsed_time_ms_bucket{le="50000"} 5
+query_total_elapsed_time_ms_bucket{le="+Inf"} 5
+query_total_elapsed_time_ms_sum 1.8407059999999997
+query_total_elapsed_time_ms_count 5
+```
+
+### Use Prisma metrics with the Prometheus metrics system
+
+In the majority of cases, Prometheus must scrape an endpoint to retrieve metrics. The following example shows how to send data with `Express.js`:
+
+<!-- prettier-ignore -->
+```js
+import { PrismaClient } from "@prisma/client";
+import express, { Request, Response } from "express";
+
+const app = express();
+const port = 4000;
+const prisma = new PrismaClient();
+
+app.get("/metrics", async (_req, res: Response) => {
+  let metrics = await prisma.$metrics.prometheus();
+  res.end(metrics);
+});
+
+app.listen(port, () => {
+  console.log(`Example app listening on port ${port}`);
+})
+```
+
+The following example shows how to combine Prisma metrics with other Prometheus client libraries, also served with an `Express.js` endpoint:
```

Comment:
```suggestion
The following example shows how to combine Prisma metrics with other Prometheus client libraries, also served with a REST API endpoint using Express.js:
```

Reply:
I'm closing the issue, but I'll bring in this suggestion, with slightly tweaked wording.